includes/chatbot/class-omise-chatbot-endpoint-controller.php

```diff
@@ -84,4 +98,92 @@ public function chatbot_facebook_verification_permission_check( $request ) {
 
 		return new WP_Error( 'omise_chatbot_failed_facebook_verification', __( 'Failed Facebook verification' ), array( 'status' => 400 ) );
 	}
+
+	/**
+	 * @param  WP_REST_Request $request  Full details about the request.
+	 *
+	 * @return true|WP_Error             True if the request has read access, otherwise WP_Error object.
+	 *
+	 * @see    https://developers.facebook.com/docs/messenger-platform/webhook
+	 *
+	 * @since  3.2
+	 */
+	public function chatbot_facebook_response( $request ) {
+		foreach ( $request['entry'] as $entry ) {
+			// Note, Facebook document says "even though this is an array, it will only contain one messaging object.".
+			// Ref. https://developers.facebook.com/docs/messenger-platform/webhook
+			$messaging_event = $entry['messaging'][0];
+
+			/**
+			 * Subscribes to Message Received events
+			 *
+			 * @see https://developers.facebook.com/docs/messenger-platform/reference/webhook-events/message
+			 */
+			if ( isset( $messaging_event['message'] ) ) {
+				// ...
+
+			/**
+			 * Postback Received events
+			 *
+			 * @see https://developers.facebook.com/docs/messenger-platform/reference/webhook-events/messaging_postbacks
+			 */
+			} else if ( isset( $messaging_event['postback'] ) ) {
+				// Event payload.
+				$sender_id = $messaging_event['sender']['id'];
+				$payload   = $messaging_event['postback']['payload'];
+
+				// TODO: This is just to prove concept, need to find other solution.
```

Comment:
Just for prove only right ?

Reply:
@igroomgrim Yes, this PR is just to prepare structure for coming request handler.
Eventually, it will likely be this: https://github.com/omise/omise-woocommerce/pull/58/files#diff-b0e25579a2040d94b4d78f2b56776d79R101

(this just to clear structure & make PR small)

Reply:
👌 Got it